src/main/kotlin/com/github/oxc/project/oxcintellijplugin/lsp/OxlintConfigWatcher.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.oxc.project.oxcintellijplugin.lsp
+
+import com.github.oxc.project.oxcintellijplugin.Constants
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.ProjectManager
+import com.intellij.openapi.vfs.newvfs.BulkFileListener
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
+import com.intellij.platform.lsp.api.LspServerManager
+
+class OxlintConfigWatcher : BulkFileListener {
+
+    override fun after(events: List<VFileEvent>) {
+        val configChanged = events.any { event ->
+            val fileName = event.path.substringAfterLast("/")
+
+            return@any Constants.CONFIG_FILES.contains(fileName)
+        }
+
+        if (configChanged) {
+            val projectManager = ApplicationManager.getApplication().service<ProjectManager>()
+            val openProjects = projectManager.openProjects
+            openProjects.forEach { project ->
```

Comment:
It's not the most efficient to restart the language server for every open project just because a config file changed in one project.  I don't think it should be that big of an issue though.

The alternative way to do this would be to identify what projects are impacted from `events` and then restart only those.  I think that doing this might require accessing the file system though (definitely in the case of `VFileCreateEvent` and possibly other events).